src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStore.java

```diff
@@ -185,18 +186,30 @@ public URL toExternalURL(@NonNull Blob blob, @NonNull HttpMethod httpMethod) thr
         String name = blob.getMetadata().getName();
         LOGGER.log(Level.FINE, "Generating presigned URL for {0} / {1} for method {2}",
                 new Object[] { container, name, httpMethod });
+        String contentType = null;
         com.amazonaws.HttpMethod awsMethod;
         switch (httpMethod) {
         case PUT:
             awsMethod = com.amazonaws.HttpMethod.PUT;
+            // Only set content type for upload URLs, so that the right S3 metadata gets set
+            contentType = blob.getMetadata().getContentMetadata().getContentType();
```

Comment:
Any chance getContentMetadata will return null?

Reply:
The constructor of the `MutableBlobMetadataImpl` class (which is mentioned in the `MutableBlobMetadata` interface via the annotation `@ImplementedBy(MutableBlobMetadataImpl.class)`) has the following implementation:

```
public MutableBlobMetadataImpl(BlobMetadata from) {
  super(from);
  this.contentMetadata = new BaseMutableContentMetadata();
  HttpUtils.copy(from.getContentMetadata(), this.contentMetadata);
  this.publicUri = from.getPublicUri();
  this.container = from.getContainer();
  this.tier = from.getTier() == null ? Tier.STANDARD : from.getTier();
}
```

So it defaults to non-null, and then could only be set to null if someone called its setter with a null parameter. The setter has no guard around this. So that probably means I should check for it?